packages/web/components/InlineFeedbackPopover/Comment.tsx

```diff
@@ -69,35 +72,75 @@ const Comment: React.FC<CommentProps> = ({ comment, canEdit, onUpdateComment, cu
     })
   }
 
-  const [createCommentThanks] = useCreateCommentThanksMutation()
-  const createNewCommentThanks = async () => {
-    await createCommentThanks({
-      variables: {
-        commentId: comment.id,
+  const [createCommentThanks, createCommentThanksResult] = useCreateCommentThanksMutation({
+    update(cache, { data }) {
+      if (data?.createCommentThanks) {
+        cache.modify({
+          id: `${comment.__typename}:${comment.id}`,
+          fields: {
+            thanks: (existingComments: CommentThanks[] = []) => {
+              return [...existingComments, data.createCommentThanks]
+            },
+          },
+        })
+      }
+    },
+    onError: (error) => toast.error(error),
+  })
+  const createNewCommentThanks = () => {
+    createCommentThanks({
+      variables: { commentId: comment.id },
+      optimisticResponse: {
+        __typename: 'Mutation',
+        createCommentThanks: {
+          __typename: 'CommentThanks',
+          id: generateNegativeRandomNumber(),
+          author: {
+            __typename: 'User',
+            handle: currentUser!.handle,
+            name: currentUser?.name,
+            profileImage: currentUser?.profileImage,
+            id: currentUser!.id,
+          },
+        },
       },
     })
-    await onUpdateComment()
   }
+  const [deleteCommentThanks, deleteCommentThanksResult] = useDeleteCommentThanksMutation({
+    update(cache, { data }) {
+      if (data?.deleteCommentThanks) {
+        cache.modify({
+          id: `${comment.__typename}:${comment.id}`,
+          fields: {
+            thanks: (existingComments: CommentThanks[] = []) => {
+              return existingComments.filter(
+                (existingComment) => data.deleteCommentThanks.id !== existingComment.id,
+              )
+            },
+          },
+        })
 
-  const [deleteCommentThanks] = useDeleteCommentThanksMutation({
-    onCompleted: () => {
-      // just refetches the post as in updateComment
-      onUpdateComment()
+        cache.evict({ id: `${data.deleteCommentThanks.__typename}:${data.deleteCommentThanks.id}` })
+      }
     },
+    onError: (error) => toast.error(error),
   })
 
   const deleteExistingCommentThanks = () => {
-    const currentCommentThanks = comment.thanks.find((thanks) => thanks.author.id === currentUserId)
+    const currentCommentThanks = comment.thanks.find(
+      (thanks) => thanks.author.id === currentUser?.id,
+    )
 
     if (currentCommentThanks) {
-      deleteCommentThanks({
-        variables: {
-          commentThanksId: currentCommentThanks.id,
-        },
-      })
+      deleteCommentThanks({ variables: { commentThanksId: currentCommentThanks.id } })
     }
   }
 
+  const isLoadingCommentThanks = useMemo(
+    () => createCommentThanksResult.loading || deleteCommentThanksResult.loading,
+    [createCommentThanksResult.loading, deleteCommentThanksResult.loading],
+  )
```

Comment:
I'd just do
```const isLoadingCommentThanks = createCommentThanksResult.loading || deleteCommentThanksResult.loading```

useMemo is going to have to do at least two comparisons and a conditional branch, which is already going to be slower than just ORing two values and function creation and call overhead adds to that too. It's a boolean so it's not like memoizing it is necessary to prevent re-renders when passing it along as a pro

Reply:
Lol! You're completely right. Sometimes I'm very addicted to memoization.
Thanks for reviewing it.